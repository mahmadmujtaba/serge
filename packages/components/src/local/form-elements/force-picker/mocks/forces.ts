const forces = [
  {
    name: 'Blue Force',
    colour: '#69c',
    selected: true
  },
  {
    name: 'Red Force',
    colour: '#f00',
    selected: false
  },
  {
    name: 'White Force',
    colour: '#fff',
    selected: false
  }
]

export default forces
