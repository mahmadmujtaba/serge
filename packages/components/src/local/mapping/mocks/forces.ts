export const forces = [
    {
      color: '#FCFBEE',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'White',
      overview: 'Umpire force.',
      roles: [
        {
          control: true,
          isInsightViewer: true,
          isObserver: true,
          name: 'Game Control',
          password: 'p2311'
        }
      ],
      umpire: true,
      uniqid: 'umpire'
    },
    {
      assets: [
        {
          name: 'HMS DEVONSHIRE',
          contactId: 'C043',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'frigate',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00001'
        },
        {
          name: 'STING',
          contactId: 'C072',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'helicopter',
          position: 'P01',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00002'
        },
        {
          name: 'P8',
          contactId: 'C076',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'fixed-wing-aircraft',
          position: 'P03',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00012'
        },
        {
          name: 'EDDYSTONE PT',
          contactId: 'C012',
          condition: 'Full capability',
          perceptions: {
            Red: {
              force: 'Blue',
              type: ''
            }
          },
          platformType: 'merchant-vessel',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00003'
        }
      ],
      color: '#00F',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Blue',
      overview: 'Blue force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p5543'
        }
      ],
      umpire: false,
      uniqid: 'Blue'
    },
    {
      assets: [
        {
          name: 'TARRADA',
          contactId: 'C065',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'fishing-vessel',
          position: 'N07',
          status: {
            state: 'Transiting',
            speedKts: 10
          },
          uniqid: 'a0pra000100'
        },
        {
          name: 'WBIED-1',
          contactId: 'C108',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T01',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra020102'
        },
        {
          name: 'WBIED-2',
          contactId: 'C118',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T03',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra130102'
        },
        {
          name: 'Osa',
          contactId: 'C003',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: ''
            }
          },
          platformType: 'fast-attack-craft',
          position: 'M05',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra160102'
        },
        {
          name: 'Yevgenya',
          contactId: 'C313',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: 'fast-attack-craft'
            }
          },
          platformType: 'mcmv',
          position: 'M04',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra170102'
        },
        {
          name: 'Coastal Radar C',
          contactId: 'C476',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'coastal-radar-site',
          position: 'Q12',
          status: {
            state: 'Inactive'
          },
          uniqid: 'a0prc000103'
        }
      ],
      color: '#F00',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Red',
      overview: 'Red force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p3244'
        }
      ],
      umpire: false,
      uniqid: 'Red'
    }
  ]


export default forces
