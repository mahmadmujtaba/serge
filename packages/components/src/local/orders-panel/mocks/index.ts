import { ADJUDICATION_PHASE } from '@serge/config'

const data = {
  selectedForce: 'blue',
  phase: ADJUDICATION_PHASE
}

export default data
