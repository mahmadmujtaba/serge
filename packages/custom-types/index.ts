import ChannelData from './channel-data.interface'
import Channel from './channel.interface'
import ForceData from './force-data.interface'
import Force from './force.interface'
import MarkerPerception from './marker-perception.interface'
import Participant from './participant.interface'
import PlatformTypeData from './platform-type-data.interface'
import PlatformType from './platform-type.interface'
import Role from './role.interface'
import States from './states.interface'
import Template from './template.interface'
import WargameData from './wargame-data.interface'
import WargameList from './wargame-list.interface'
import WargameOverview from './wargame-overview.interface'
import Wargame from './wargame.interface'

export {
  ChannelData,
  Channel,
  ForceData,
  Force,
  MarkerPerception,
  Participant,
  PlatformTypeData,
  PlatformType,
  Role,
  States,
  Template,
  WargameData,
  WargameList,
  WargameOverview,
  Wargame
}