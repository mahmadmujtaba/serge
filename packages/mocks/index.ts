/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as MessagesMock } from './messages.mock'
export { default as WargameMock } from './wargame.mock'