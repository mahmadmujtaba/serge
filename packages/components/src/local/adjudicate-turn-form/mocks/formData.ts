import { AdjudicateTurnFormPopulate } from "@serge/custom-types"

const formData: AdjudicateTurnFormPopulate = {
  contactId: 'C234',
  status: [
    {
      name: 'Fishing',
      mobile: false
    },
    {
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }
  ],
  speed: [10, 20, 30],
  visibleTo: [
    {
      name: 'Blue Force',
      colour: '#69c'
    },
    {
      name: 'Red Force',
      colour: '#f00'
    },
    {
      name: 'White Force',
      colour: '#fff'
    }
  ],
  condition: ['Working', 'Disabled', 'Immobile', 'Destroyed']
}

export default formData
