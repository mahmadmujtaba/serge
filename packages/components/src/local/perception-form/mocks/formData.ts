import { PerceptionFormData } from '@serge/custom-types'

const formData: PerceptionFormData = {
  populate: {
    perceivedForces: [
      {
        name: 'Blue Force',
        colour: '#69c'
      },
      {
        name: 'Red Force',
        colour: '#f00'
      },
      {
        name: 'White Force',
        colour: '#fff'
      }
    ],
    perceivedTypes: [
      { uniqid: 'aa', name:'Fishing vessel'},
      { uniqid: 'a1', name:'torpedo'},
      { uniqid: 'a2', name:'Boghammer'},
      { uniqid: 'a3', name:'Frigate'}
    ]
  },
  values: {
    perceivedNameVal: 'Test',
    perceivedForceVal: 'blue-force',
    perceivedTypeId: 'a3',
    assetId: 'c00012',
    iconURL: 'frigate.svg'
  }
}

export default formData
