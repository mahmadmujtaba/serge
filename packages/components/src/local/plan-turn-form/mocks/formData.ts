import { COMMODITY_TYPE_NUMBER, COMMODITY_VALUE_NUMBER } from '@serge/config'
import { PlanTurnFormData } from '@serge/custom-types'

const formData: PlanTurnFormData = {
  populate: {
    status: [{
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }],
    speed: [10, 20, 30],
    attributes: [{ commId: 'comm_a', commType: COMMODITY_TYPE_NUMBER, name: 'Fuel', editableByPlayer: false },
      { commId: 'comm_b', commType: COMMODITY_TYPE_NUMBER, name: 'People', editableByPlayer: false },
      { commId: 'comm_c', commType: COMMODITY_TYPE_NUMBER, name: 'Water', units: 'litres', editableByPlayer: false }]
  },
  values: {
    statusVal: {
      name: 'Transiting',
      mobile: true
    },
    speedVal: 10,
    turnsVal: 5,
    condition: 'Working',
    attributes: [{ commId: 'comm_a', commType: COMMODITY_VALUE_NUMBER, value: 12 },
      { commId: 'comm_b', commType: COMMODITY_VALUE_NUMBER, value: 213 },
      { commId: 'comm_c', commType: COMMODITY_VALUE_NUMBER, value: 12450 }]
  }
}

export default formData
