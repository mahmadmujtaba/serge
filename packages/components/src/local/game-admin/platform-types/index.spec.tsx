/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import PlatformTypes from './index'

import { platformType } from '@serge/mocks'
import { PlatformType } from '@serge/custom-types'

const onChange = (platformType: PlatformType): void => {
  console.log(platformType)
}

it('platform-types renders correctly', () => {
  const tree = renderer
    .create(<PlatformTypes
      platformType={platformType}
      onChange={onChange}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
