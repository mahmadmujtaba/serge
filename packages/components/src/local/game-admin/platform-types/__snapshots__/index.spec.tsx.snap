// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`platform-types renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="left"
  >
    <div
      className="main"
    >
      <div
        className="section"
      >
        <button
          className="main"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </div>
      <ul />
    </div>
  </div>
  <div
    className="right"
  >
    <a
      className="main"
      href="#"
      onClick={[Function]}
    >
      Text
    </a>
  </div>
</div>
`;
